content/docs/guides/configuration.md

```diff
@@ -25,24 +26,65 @@ The following is an example of the `dataform.json` file for a BigQuery project:
 }
 ```
 
+All of these configuration settings are accessible in your project code as properties of the `dataform.projectConfig` object. For example:
+
+```js
+// definitions/my_view.sqlx
+config { type: "view" }
+select ${when(
+  dataform.projectConfig.warehouse === "bigquery",
+  "warehouse is set to bigquery!",
+  "warehouse is not set to bigquery!"
+)}
+```
+
 ### Configure default schema names
 
 Dataform aims to create all objects under a single schema (or dataset in BigQuery) in your warehouse. This is usually called `dataform` but can be changed
 by changing the `defaultSchema` property to some other value. For example, to change it to `mytables`, update the configuration file as following:
 
 ```json
+// dataform.json
 {
   ...
   "defaultSchema": "mytables",
   ...
 }
 ```
 
+### Configure custom compilation variables
+
+You may inject custom variables into project compilation:
+
+```json
+// dataform.json
+{
+  ...
+  "variables": {
```

Comment:
Sorry I meant to suggest this earlier. How about just `vars` ?
Don't feel strongly, makes everything a little shorter and just as clear IMO?